spec/unit/config_spec.rb

```diff
@@ -25,4 +25,10 @@
       end
     end
   end
+
+  describe ".intercept_401_responses" do
+    it "defaults to true" do
+      expect(subject.intercept_401_responses).to eq(true)
+    end
```

Comment:
Hmm it feels like just this isn't really testing this config option sufficiently. Perhaps we need a test in the railtie file that asserts that this config option actually does something. What do you think?

Reply:
hmm yup good point, are you thinking of a test that we can set this value in a similar manner to the example of the description: `GDS::SSO.config { |config| config.intercept_401_responses = false }` ?

Reply:
More just something that asserts that this config value results in warden having intercept_401 switched on or not.

The sort of thing I'd want the test to verify is that if warden ever dropped the feature or renamed it this test would catch it.

Reply:
i've added a testing approach, couldn't find a more elegant way, otherwise i would need to make an integration test.

Reply:
Came up with this, what do you think?

```
  describe "configuring intercept_401_responses" do
    it "sets warden intercept_401 to false when the configuration option is set to false" do
      GDS::SSO::Config.intercept_401_responses = false

      described_class.initializers.each(&:run)

      expect(warden_manager.config[:intercept_401]).to be(false)
    end

    it "sets warden intercept_401 to true when the configuration option is set to true" do
      GDS::SSO::Config.intercept_401_responses = true

      described_class.initializers.each(&:run)

      expect(warden_manager.config[:intercept_401]).to be(true)
    end
  end

  def warden_manager
    middleware = Rails.application.config.middleware.find { |m| m.name.include?("Warden::Manager") }

    Warden::Manager.new(nil, &middleware.block)
  end

```

Reply:
I set it up in a loop to run 1000 times and it didn't flake at all so I think it's ok 🤞

Reply:
that's strange for me it does it like every 2-3 runs, 

<img width="761" alt="image" src="https://github.com/alphagov/gds-sso/assets/8425834/e8988d68-770b-4332-9078-e2c2b372e3d1">

Reply:
Though I see your code is perhaps different given the use of subject 😕

Reply:
it's the `spec/unit/config_spec.rb` that failing, line 29. I suspect it's the random seed, it depends what test runs last it affects what the outcome is for that spec

Reply:
Ahh - sorry. I'd misunderstood that. 

Yes it makes sense that this one is flaky given that just has global vars. I don't really think the config test is sufficiently valuable to keep.

Though it is annoying to have tests that affect global state, so lets change the tests I suggested to mocks. e.g: 

```
        allow(GDS::SSO::Config).to receive(:intercept_401_responses).and_return(false)
```